Allow external tags instead of fetching them from git (#354)
jupyterhub/action-major-minor-tag-calculator: README.md

[![build-test](https://github.com/jupyterhub/action-major-minor-tag-calculator/workflows/build-test/badge.svg)](https://github.com/jupyterhub/action-major-minor-tag-calculator/actions)

> Intelligently tag a build artifact starting from a semantic version tag (`major.minor.patch`), and breaking in into partial components (`major,.minor`, `major`), taking into account existing partial tags.

Reviewer: ```suggestion
Intelligently tag a build artifact starting from a semantic version tag (`major.minor.patch`), and breaking in into partial components (`major`, `minor`, `major`), taking into account existing partial tags.
```

> Intelligently tag a build artifact starting from a semantic version tag (`major.minor.patch`), and breaking in into partial components (`major,.minor`, `major`), taking into account existing partial tags.

Author: ```suggestion
Intelligently tag a build artifact starting from a semantic version tag (`major.minor.patch`), and breaking it into partial components (i.e. `major.minor`, and `major`), taking into account existing partial tags.
```

## Recommended input parameters

By default this action works with Git tags. It will fetch tags from the current repository, and if this action is triggered by a tag push it will autodetect the current tag.

- `githubToken`: The GitHub token, required so this action can fetch tags using the GitHub API. If this parameter is not set then `defaultTag` (if set) or an empty list will be returned.

> You can use this action with non-Git tags by explicitly providing the list of existing tags, and the current (new) tag.

Reviewer: ```suggestion
You can use this action with non-Git tags by explicitly providing the list of existing tags, and the new tag.
```

In this case you should omit `githubToken` as no GitHub API calls are required.

- `existingTags`: List of existing tags as a JSON array, provide this if you aren't processing Git tags, e.g. `["1.2.3-0", "1.2.3-1"]`

> - `newTag`: The current (new) tag, provide this if you aren't processing Git tags, e.g. `"1.2.4-0"`

Reviewer: ```suggestion
- `newTag`: The new tag, provide this if you aren't processing Git tags, e.g. `"1.2.4-0"`
```
